core/client/fs/src/main/java/alluxio/client/block/BlockMasterClientPool.java

```diff
@@ -38,14 +36,12 @@
   /**
    * Creates a new block master client pool.
    *
-   * @param context the information required for connecting to Alluxio
-   * @param masterInquireClient a client for determining the master address
+   * @param ctx the information required for connecting to Alluxio
    */
-  public BlockMasterClientPool(ClientContext context, MasterInquireClient masterInquireClient) {
-    super(context.getConf().getInt(PropertyKey.USER_BLOCK_MASTER_CLIENT_THREADS));
+  public BlockMasterClientPool(MasterClientContext ctx) {
+    super(ctx.getClusterConf().getInt(PropertyKey.USER_BLOCK_MASTER_CLIENT_THREADS));
     mClientList = new ConcurrentLinkedQueue<>();
-    mMasterContext =
-        MasterClientContext.newBuilder(context).setMasterInquireClient(masterInquireClient).build();
+    mMasterContext = ctx;
```

Comment:
@ZacBlanco instead of a copy of `ClientContext` per pool, we are not only making a copy in `FileSystemContext`. this should be okay as the context passed by the client is still copied. PTAL